src/webgpu/util/copy_to_texture.ts

```diff
@@ -165,31 +236,93 @@ export class CopyToTextureUtils extends GPUTest {
 
   // MAINTENANCE_TODO(crbug.com/dawn/868): Should be possible to consolidate this along with texture checking
   checkBufferWithRowPitch(
-    actual: Uint8Array,
-    exp: Uint8Array,
+    actual: TypedArrayBufferView,
+    expected: TypedArrayBufferView,
     width: number,
     height: number,
     rowPitch: number,
     bytesPerPixel: number,
-    isFp16: boolean
+    dstFormat: RegularTextureFormat
   ): string | undefined {
     const bytesPerRow = width * bytesPerPixel;
+
     // When dst format is fp16 formats, the expectation and real result always has 1 bit difference in the ending
     // (e.g. CC vs CD) if there needs some alpha ops (if alpha channel is not 0.0 or 1.0). Suspect it is errors when
     // doing encoding. We check fp16 dst texture format with 1-bit ULP tolerance.
-    if (isFp16) {
+    if (isFp16Format(dstFormat)) {
+      const expF16bits = new Uint16Array(
+        expected.buffer,
+        expected.byteOffset / Uint16Array.BYTES_PER_ELEMENT,
+        expected.byteLength / Uint16Array.BYTES_PER_ELEMENT
+      );
+      const checkF16bits = new Uint16Array(
+        actual.buffer,
+        actual.byteOffset / Uint16Array.BYTES_PER_ELEMENT,
+        actual.byteLength / Uint16Array.BYTES_PER_ELEMENT
+      );
+
+      for (let y = 0; y < height; ++y) {
+        const expRowF16bits = expF16bits.subarray(
+          (y * bytesPerRow) / expF16bits.BYTES_PER_ELEMENT,
+          bytesPerRow / expF16bits.BYTES_PER_ELEMENT
+        );
+
+        const checkRowF16bits = checkF16bits.subarray(
+          (y * bytesPerRow) / checkF16bits.BYTES_PER_ELEMENT,
+          bytesPerRow / checkF16bits.BYTES_PER_ELEMENT
+        );
+
+        const expRowF32bits = new Float32Array(expRowF16bits.length);
+        const checkRowF32bits = new Float32Array(checkRowF16bits.length);
```

Comment:
These aren't the bits for f32 values, they're actual f32 values. Should just be expRowF32/checkRowF32.

Reply:
It seems I mixed f32 and f32bits to the same meaning as the bit length. Thanks !

Reply:
Ah, you're right it's very confusing!